crates/db/src/stubs/chain_state.rs

```diff
@@ -88,10 +88,10 @@ impl ChainstateDatabase for StubChainstateDb {
 
         // Remove from the two tables.  This does have to touch every state in
         // the table but it's fine because this will never be used in production.
-        let states_removed = st.toplevels.extract_if(|idx, _| *idx < before_idx).count();
+        let states_removed = st.toplevels.extract_if(0..before_idx, |_, _| true).count();
```

Comment:
Isn't this just a more complicated split?

Reply:
That was an automated `cargo clippy --fix`

Reply:
Hm, I guess it's not quite that simple, since it's a `BTreeMap`.